.github/workflows/bazel.yml

```diff
@@ -183,3 +183,12 @@ jobs:
           run: |
             cd /workspaces/magma
             bazel test //... --test_output=errors --cache_test_results=no
+      - name: Run `bazel run //:check_starlark_format`
```

Comment:
yeah I haven't had a chance to look into this, but `check_starlark_format` **should** be run as part of bazel test //... but it doesn't seem to catch all issues. Is this identical as running `bazel run //:buildifier` and then checking for modified files?

Reply:
My understanding:
* `buildifier_test` will supersede `buildifier` with `mode` `diff` (deprecated)
* I would have also expect that a `buildifier_test` target should be run with `bazel test`, but I was not able to get the target to fail reliable with `test`. E.g., formmatting issues in `$MAGMA_ROOT/BUILD.bazel` are detected via `bazel test`. But not in other build files. `bazel run` seems to be reliable.

> Is this identical as running bazel run //:buildifier and then checking for modified files?

This is my understanding. I.e., same as `bazel run //:buildifier` with `mode` `diff`

Reply:
Thanks for the summary :) so is the difference triggering it with run vs test?

Reply:
yes, run works in my experiments - but the test also seems to fail for linter warnings. I will have an additional look. I could try to remove the linting from the check - do you still need/use the `check_starlark_format` in it's current state? If yes, I can create a new one without the linting.
FYI: the linter warnings are semi-helpful. The complaints seem to be for starlark syntax in general, i.e., bazel specific syntax is also in the warnings, e.g., unused name argument in custom rules.

Reply:
Yes let's address the lint warnings and keep it on. We only seem to have a few!

Reply:
ok, with the second attempt I now understand how the exclude mechanics work :)
-> excluded the unused variable lint warning and fixed doc string warnings

I still see behavior in the buildifier that I do not understand, e.g., needed function doc string in `go_repositories.bzl` but not in `python_repositories.bzl`, no property ordering in `go_repositories.bzl`, ...
But I think this change is still a valid improvement.